src/posix/platform/system.cpp

```diff
@@ -183,6 +192,7 @@ otInstance *otSysInit(otPlatformConfig *aPlatformConfig)
 
     platformInit(aPlatformConfig);
 
+    SettingsSetDryRun(aPlatformConfig->mDryRun);
```

Comment:
Save `mDryRun` as a posix platform scope variable so that all posix components can access?
Otherwise we may end up setting `DryRun` for a number of components.

Reply:
I agree. Currently only the settings module need to access it. I think we will need to add a method for other components to access `DryRun`. What place do you think is good to put this variable? We can also put it in `settings.cpp` and add a method for other components to access it. Thoughts?

Reply:
We already have global variable `gInstance` in system.cpp that is accessiable by all posix modules. 
Should we follow the same convention?

Reply:
Updated!